### ISSUE TITLE
How should we set the network to `training` mode ?

### ISSUE THREAD

**Issue by jrabary** [tensorflow/agents#6]
When we have layers with different behaviour in train and test phase (e.g BatchNormalization) inside the network we need to set this mode explicitly. Coming from `tf.keras.Model` we usually set this in the `call` method of the network.
How can we achieve the same thing with tf_agent ? Currently the API of the `call` function in the subclasses of `network.Network` seems to not offer such phase setting.

**Comment by oars** [comment]
Don't think we have any networks being used currently that require a phase to be set.

For now you could add a new `set_phase` method to your network and manually change it in the train_eval.py binary when evaluations start.
